src/SessionSetup/ServiceDeployManager.cpp

```diff
@@ -449,78 +452,106 @@ ErrorMessageOr<void> ServiceDeployManager::CopyOrbitUserSpaceInstrumentationLibr
   return outcome::success();
 }
 
-ErrorMessageOr<void> ServiceDeployManager::StartOrbitService() {
-  ORBIT_CHECK(QThread::currentThread() == thread());
-  emit statusMessage("Starting OrbitService on the remote instance...");
+[[nodiscard]] static std::string GenStartOrbitServiceCommand(
```

Comment:
Yes, right. Could be in an anonymous namespace. I personally prefer `static` because it's so much easier to spot. I can't count anymore how often I asked someone to change the linkage of a free function to internal without realizing that it was already in an anonymous namespace. Furthermore I personally prefer to have this function close by to the one instance that is using it. Putting it at the top of the file just add an unecessary burden for the reader. If it was used in multiple places, then of course. But this function only exists to give this part of the code a name.

Despite all of that, I don't feel strongly about any of those. Would you prefer me to move the function up into the anonymous namespace?

Reply:
Interesting opinion, I have not thought about this in terms of locality in a file. I dont feel strongly about this either. I think you should leave it as is